src/agent/ncp_openthread.cpp

```diff
@@ -322,6 +328,37 @@ void ControllerOpenThread::RegisterResetHandler(std::function<void(void)> aHandl
     mResetHandlers.emplace_back(std::move(aHandler));
 }
 
+void ControllerOpenThread::HandleRegionCommand(void *aContext, uint8_t aArgLength, char **aArgs)
+{
+    ControllerOpenThread *controller = static_cast<ControllerOpenThread *>(aContext);
+    controller->HandleRegionCommand(aArgLength, aArgs);
+}
+
+void ControllerOpenThread::HandleRegionCommand(uint8_t aArgLength, char **aArgs)
+{
+    if (aArgLength == 0)
+    {
+        otCliOutputFormat("%s\n", mRegionCode.c_str());
+        otCliOutputFormat("Done\n", mRegionCode.c_str());
+    }
+    else if (aArgLength == 1)
+    {
+        if (strnlen(aArgs[0], 3) == 2)
+        {
+            mRegionCode = aArgs[0];
+            otCliOutputFormat("Done\n");
+        }
+        else
+        {
+            otCliOutputFormat("Error: InvalidArgs\n");
```

Comment:
not in this PR, but prefer change the prototype of otCliCommand to return otError and handle error message in OpenThread. Just like the diagnostic commands.